src/uvarint.cr

```diff
@@ -1,44 +1,131 @@
+require "big/big_int"
+
 require "./uvarint/*"
 
 # This module is based on the varint implementation in the Go programming
 # language. See: https://golang.org/src/encoding/binary/varint.go
 
-MSB = 0x80_u8  # 10000000
-REST = 0x7F_u8 # 01111111
-
-module UVarint
-  extend self
+private MSB = 0x80_u8
+private REST = 0x7F_u8
+private ZERO = 0.to_big_i
 
-  alias UVarInt = Array(UInt8)
+# Encodes an Int::Unsined into a Bytes slice.
+private def encode(n : Int::Unsigned | BigInt) : Bytes
+  arr = [] of UInt8
+  while n >= MSB
+    arr << (n.to_u8 | MSB)
+    n = n >> 7
+  end
+  arr << n.to_u8
+  ptr = arr.to_unsafe
+  Bytes.new(ptr, arr.size)
+end
 
-  # encode encodes a UInt64 into a UVarInt,
-  # which is an Array of UInt8's.
-  def encode(n : UInt64) : UVarInt
-    buf = [] of UInt8
-    while n >= MSB
-      buf << (n.to_u8 | MSB)
-      n = n >> 7
+# Decodes enumerable bytes into a BigInt.
+# If the enumeration is longer thar 10 bytes,
+# then an overflow exception is raised.
+private def decode(bytes : Bytes) : BigInt
+  x = ZERO
+  s = 0
```

Comment:
I agree, that is confusing. For reference, `ZERO == BigInt(0)`. What would be a better name here? `x` is the `BigInt` value that is decoded into.